lumeer-storage/lumeer-storage-mongodb/src/main/java/io/lumeer/storage/mongodb/dao/system/MongoGroupDao.java

```diff
@@ -0,0 +1,117 @@
+/*
+ * Lumeer: Modern Data Definition and Processing Platform
+ *
+ * Copyright (C) since 2017 Answer Institute, s.r.o. and/or its affiliates.
+ *
+ * This program is free software: you can redistribute it and/or modify
+ * it under the terms of the GNU General Public License as published by
+ * the Free Software Foundation, either version 3 of the License, or
+ * (at your option) any later version.
+ *
+ * This program is distributed in the hope that it will be useful,
+ * but WITHOUT ANY WARRANTY; without even the implied warranty of
+ * MERCHANTABILITY or FITNESS FOR A PARTICULAR PURPOSE.  See the
+ * GNU General Public License for more details.
+ *
+ * You should have received a copy of the GNU General Public License
+ * along with this program.  If not, see <http://www.gnu.org/licenses/>.
+ */
+package io.lumeer.storage.mongodb.dao.system;
+
+import static io.lumeer.storage.mongodb.util.MongoFilters.idFilter;
+
+import io.lumeer.api.model.Group;
+import io.lumeer.storage.api.dao.GroupDao;
+import io.lumeer.storage.api.exception.StorageException;
+import io.lumeer.storage.mongodb.codecs.GroupCodec;
+import io.lumeer.storage.mongodb.model.MongoGroup;
+
+import com.mongodb.MongoException;
+import com.mongodb.client.MongoCollection;
+import com.mongodb.client.model.Filters;
+import com.mongodb.client.model.FindOneAndUpdateOptions;
+import com.mongodb.client.model.IndexOptions;
+import com.mongodb.client.model.Indexes;
+import com.mongodb.client.model.ReturnDocument;
+import com.mongodb.client.model.Updates;
+import com.mongodb.client.result.DeleteResult;
+
+import org.bson.Document;
+import org.bson.conversions.Bson;
+
+import java.util.ArrayList;
+import java.util.List;
+import javax.annotation.PostConstruct;
+import javax.enterprise.context.RequestScoped;
+
+@RequestScoped
+public class MongoGroupDao extends SystemScopedDao implements GroupDao {
```

Comment:
I thought we made it clear that groups will be organization-scoped. Or are there any performance benefits or other reasons why it is better to have it system-scoped? Because I cannot see any at the moment.

Reply:
Each organization will contain only few groups. Do we really want to have so many small collections?

Reply:
Good point. But how do you then want to deal with unique group names just within a single organization if all groups are stored in a single collection?

Reply:
It's simple: just add compound unique index on `organizationId` and `groupName`. It's also tested. (look at `MongoGroupDaoTest#testCreateGroupAnotherOrganization` method)

Reply:
And what about our idea of storing organization data in different MongoDB databases and having only really necessary collections in the system database?